src/script/sign.cpp

```diff
@@ -103,12 +103,28 @@ bool Solver(const CKeyStore& keystore, const CScript& scriptPubKey, uint256 hash
     case TX_MULTISIG:
         scriptSigRet << OP_0; // workaround CHECKMULTISIG bug
         return (SignN(vSolutions, keystore, hash, nHashType, scriptSigRet));
+
+    case TX_COLDSTAKE:
+        if (fColdStake) {
+            // sign with the cold staker key
+            keyID = CKeyID(uint160(vSolutions[0]));
+        } else {
+            // sign with the owner key
+            keyID = CKeyID(uint160(vSolutions[1]));
+        }
+        if (!Sign1(keyID, keystore, hash, nHashType, scriptSigRet))
+            return error("*** %s: failed to sign with the %s key.",
+                    __func__, fColdStake ? "cold staker" : "owner");
+        CPubKey vch;
+        keystore.GetPubKey(keyID, vch);
```

Comment:
Yes, the old code has the same in `TX_PUBKEYHASH` case. Better to fix that as well since we are at it.
Done in https://github.com/PIVX-Project/PIVX/pull/955/commits/47553430c582adfdefcea0babae7d1624f439459 .